scripts/update.sh

```diff
@@ -0,0 +1,193 @@
+
+    else
+        messeage "ERROR: Failed to call 'git', are you sure it is installed." "red"
+        exit 1
+    fi
+}
+
+# A function for updating MOOSE
+function update_moose()
+{
+    message "Updating MOOSE"
+
+    # If MOOSE_DIR is set
+    if [ ! -z "$MOOSE_DIR" ]; then
+
+        # Incidate that a user-defined MOOSE_DIR is being used
+        message "WARNING: MOOSE_DIR set - overriding default installed path" "yellow"
+        cd ${MOOSE_DIR}
+
+        # Determine the current git branch for MOOSE
+        BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+        if [[ "${BRANCH}" != "master" && "${BRANCH}" != "devel" ]]; then
+            message "ERROR: You are the '${BRANCH}' branch within the MOOSE repository, only the 'devel' or 'master' branches support automatic updates" "red"
+        else
+            message "  Updating MOOSE repository..." "green"
+            git pull --rebase git@github.com:idaholab/moose.git ${BRANCH}
+        fi
+
+    # Not using MOOSE_DIR, update the submodule within the application
+    else
+        export MOOSE_DIR=${APPLICATION_DIR}/moose
+        update_submodule ${APPLICATION_DIR} "moose"
+    fi
+
+    message "  Using MOOSE at ${MOOSE_DIR}" "green"
+    cd ${APPLICATION_DIR}
+}
+
+# A function for updating libMesh
+function update_libmesh()
+{
+    message "Updating libMesh"
+
+    # If LIBMESH_DIR is set
+    if [ ! -z "$LIBMESH_DIR" ]; then
+
+        # Indicate that the user-defined LIBMESH_DIR is being used
+        message "WARNING: LIBMESH_DIR set - overriding default installed path" "yellow"
+        message "         No cleaning will be done in specified path" "yellow"
+        mkdir -p $LIBMESH_DIR
+
+        cd ${LIBMESH_DIR}
+
+        # Determine the current git branch for LIBMESH
+        BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+        if [[ "${BRANCH}" != "master" ]]; then
+            message "ERROR: You are the '${BRANCH}' branch within the libMesh repository, only the 'master' branch supports automatic updates" "red"
+            exit 1
+        else
+            message "  Updating libMesh repository..." "green"
+            git pull --rebase git@github.com:libMesh/libmesh.git ${BRANCH}
+        fi
+
+    # If not using LIBMESH_DIR, update the submodule
+    else
+        export LIBMESH_DIR=${MOOSE_DIR}/libmesh/installed
+        if [ ! -d "${LIBMESH_DIR}" ]; then
+            cd ${MOOSE_DIR}
+            update_submodule ${MOOSE_DIR} "libmesh"
+            mkdir ${LIBMESH_DIR}
+
+        elif [ -d "${LIBMESH_DIR}" ]; then
+            cd ${MOOSE_DIR}/libmesh
+            rm -rf installed
+            cd - >/dev/null # Make this quiet
+            update_submodule ${MOOSE_DIR} "libmesh"
+
+        else
+            message "  Failed to update libMesh!" "red"
+            exit 1
+        fi
+    fi
+
+    message "  Using libMesh at ${LIBMESH_DIR}" "green"
+    cd ${APPLICATION_DIR}
+}
+
+# Configure and compile libMesh
+function build_libmesh()
```

Comment:
This function is a duplicate of the one in `update_and_rebuild_libmesh.sh`. Can't we just import the existing method?